File: Modules/SEDO/EMS/SIE.Web.EMS/Equipments/Units/Commands/AddUnitCommand.js
SIE.defineCommand('SIE.Web.EMS.Equipments.Units.Commands.AddUnitCommand', {
    meta: { text: "添加".t(), group: "edit", iconCls: "iconfont icon-AddEntity icon-green" },
    canExecute: function (view) {
        var isListView = view.isListView;
        if (isListView) {
            return view.canAddItem();
        }
        else
            return this.callParent();
    },
    execute: function (view, source) {
        var parent = view._getTreeRoot();
        var rootNode = view._createTreeNode(parent);
        result = parent.insertChild(0, rootNode);
        result.generateId();
        result.setTreePId(null);
        view.getControl().setSelection(result);
    },
});

File: Modules/SEDO/EMS/SIE.Web.EMS/Equipments/Units/Commands/AddUnitCommand.cs
using SIE.Web.Command;

namespace SIE.Web.EMS.Equipments.Units.Commands
{
    /// <summary>
    /// 添加设备单元命令
    /// </summary>
    [JsCommand("SIE.Web.EMS.Equipments.Units.Commands.AddUnitCommand")]
    public class AddUnitCommand : ViewCommand<ViewArgs>
    {
        /// <summary>
        /// 执行命令
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return true;
        }
    }
}
